src/vm/threads.cpp

```diff
@@ -7942,6 +7952,31 @@ void Thread::DeleteThreadStaticData(ModuleIndex index)
     m_ThreadLocalBlock.FreeTLM(index.m_dwIndex, FALSE /* isThreadShuttingDown */);
 }
 
+void* Thread::AllocTailCallArgBuffer(size_t size, void* gcDesc)
+{
+    m_tailCallTls.ArgBufferSize = size;
```

Comment:
It seems it would be nicer to encapsulate the allocation / freeing logic inside of the TailCallTls. Its members can be private then.

Reply:
`TailCallTls` is a shared structure with C# so I would prefer not to do that. Also the arg buffer will end up pointing into the static buffer in `Thread` as it stands currently.

Reply:
I don't see a problem in the native struct having methods that the managed one doesn't have. All that matters is that they have the same layout. And the m_inlineTailCallArgBuffer can also be moved into the TailCallTls for even better encapsulation. We do a similar thing for managed LoaderAllocator and its native counterpart (LoaderAllocatorObject)

Reply:
Ok -- I can do that -- if I put those fields at the end I guess I can avoid having to leak the details (like the size of the static buffer) into the managed struct?

Reply:
I do not mind fixing it.